File: Cc.Ui/src/models/appointment.ts
import { Entity, resource, field, required } from '../framework/entity';
import { Pagination, Pager } from '../framework/core/actions';

@resource('appointment')
export class Appointment extends Entity {

    @field()
    id: number;
    @field()
    eventId: number;
    @field()
    appointmentTypeId: number;
    @field()
    appointmentEndingId: number;
    @field()
    motivation: string;
    @field()
    salesmanId: number;
    @field()
    customerId: number;
    @field()
    district: string;
    @field()
    productId: number;
    @field()
    dateFrom: string;
    @field()
    dateTo: string;
    @field()
    state: number;
    @field()
    insertUser: string;
    @field()
    insertDate: string;
    @field()
    updateUser: string;
    @field()
    updateDate: string;
    @field()
    title?: string;

    constructor(data?: any) {
        super()
        this.init(data);      
    }

    static newItem(): Appointment {

        let item = new Appointment();
        return item;
    }
}

@resource('appointment')
export class AppointmentSearchModel extends Entity implements Pagination {
    
    @field()
    eventId?: number;
    @field()
    appointmentTypeId?: number;
    @field()
    appointmentEndingId?: number;
    @field()
    motivation?: string;
    @field()
    salesmanId?: number;
    @field()
    customerId?: number;
    @field()
    district?: string;
    @field()
    productId?: number;
    @field()
    dateFrom?: string;
    @field()
    dateTo?: string;
    @field()
    state?: number;

    @field()
    pager: Pager = <Pager> {
        take: 100,
        orderBy: 'id'
    };
}

File: Cc.Core/Models/Appointment.cs
using System;
using System.Collections.Generic;

namespace Cc.Core.Models
{
    public partial class Appointment
    {
        public long Id { get; set; }
        public long EventId { get; set; }
        public long AppointmentTypeId { get; set; }
        public long? AppointmentEndingId { get; set; }
        public string? Motivation { get; set; }
        public long SalesmanId { get; set; }
        public long CustomerId { get; set; }
        public string District { get; set; } = null!;
        public long? ProductId { get; set; }
        public DateTime? DateFrom { get; set; }
        public DateTime? DateTo { get; set; }
        public byte? State { get; set; }
        public string? InsertUser { get; set; }
        public DateTime? InsertDate { get; set; }
        public string? UpdateUser { get; set; }
        public DateTime? UpdateDate { get; set; }
        public string? Title { get; set; }

        public virtual AppointmentEnding? AppointmentEnding { get; set; }
        public virtual AppointmentType AppointmentType { get; set; } = null!;
        public virtual Principal Salesman { get; set; } = null!;
    }
}
